controller/linodemachine_controller.go

```diff
@@ -240,13 +240,10 @@ func (r *LinodeMachineReconciler) reconcile(
 
 		failureReason = cerrs.UpdateMachineError
 
-		if machineScope.LinodeMachine.Spec.InstanceID != nil {
-			logger = logger.WithValues("ID", *machineScope.LinodeMachine.Spec.InstanceID)
-		}
-
 		res, linodeInstance, err = r.reconcileUpdate(ctx, logger, machineScope)
-
-		return
+		if linodeInstance != nil || err == nil {
+			return
+		}
```

Comment:
Could you please add a comment to explain this block? What happens if `err != nil`? Why do we want to stop reconciliation here? Could we simply call `return r.reconcileUpdate(...)` instead?

Reply:
Added a comment